HW2/P5/physics.pyx

```diff
@@ -113,18 +128,25 @@ cpdef update(FLOAT[:, ::1] XY,
         #
         # SUBPROBLEM 1: parallelize this loop over 4 threads, with static
         # scheduling.
-        for i in range(count):
-            sub_update(XY, V, R, i, count, Grid, grid_spacing)
 
+        for i in prange(count, num_threads=4, schedule="static", chunksize=chunk):
+            sub_update(XY, V, R, i, count, Grid, grid_spacing, locks)
+        
         # update positions
         #
         # SUBPROBLEM 1: parallelize this loop over 4 threads (with static
         #    scheduling).
         # SUBPROBLEM 2: update the grid values.
-        for i in range(count):
+
+        for i in prange(count, num_threads=4, schedule="static", chunksize=chunk):
+            grid_x = int(XY[i,0] / grid_spacing)
+            grid_y = int(XY[i,1] / grid_spacing)
+            Grid[grid_x, grid_y] = -1
             for dim in range(2):
                 XY[i, dim] += V[i, dim] * t
-
+            new_x = int(XY[i,0] / grid_spacing)
```

Comment:
is it possible for the new XY co-ordinate to be off the grid at this point in the code?